qemu/tests/qemu_guest_agent.py

```diff
@@ -181,14 +183,31 @@ def gagent_verify(self, params, vm):
         self.gagent.verify_responsive()
         logging.info(self.gagent.cmd("guest-info"))
 
+    @error_context.context_aware
     def setup(self, test, params, env):
         BaseVirtTest.setup(self, test, params, env)
         start_vm = params["start_vm"]
         if (not self.vm) and (start_vm == "yes"):
             vm = self.env.get_vm(params["main_vm"])
             vm.verify_alive()
             self.vm = vm
-            self.setup_gagent_in_guest(params, self.vm)
+
+            error_context.context("Check qemu-guest-agent status before install it", logging.info)
+            session = self._get_session(params, vm)
+            s, o = session.cmd_status_output(params.get("gagent_status_cmd"))
+            if s:
+                if params["os_type"] == "windows":
```

Comment:
Not suggest to add this statement, since we already has a child class for windows.

Reply:
Agree. I seperate it in different class.
Thanks.